service/history/task/transfer_active_task_executor.go

```diff
@@ -364,6 +379,17 @@ func (t *transferActiveTaskExecutor) processCloseExecution(
 		return err
 	}
 
+	// TODO: pass crossClusterTaskGenerators to processParentClosePolicy to add new cross
+	// cluster tasks then move the call below after processParentClosePolicy call
+	if len(crossClusterTaskGenerators) > 0 {
+		t.generateCrossClusterTasks(
```

Comment:
Added the error handling. 

I was planning move this down after the parent close policy changes. Could be better to keep the changes smaller and for one API. In the PR after this one it will be after the `processParentClosePolicy`.